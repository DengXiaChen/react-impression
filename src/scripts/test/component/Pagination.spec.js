import React from 'react';
import { shallow } from 'enzyme';
import { Pagination } from '../../components/impression';

const setup = () => {
    const props = {
        activePage: 1,
        scope: 3,
        ellipsis: true,
        totalPage: 10,
        onSelect: jest.fn(),
    };

    const wrapper = shallow(<Pagination {...props} />);

    return { props, wrapper };
};

describe('Pagination', () => {
    const { props, wrapper } = setup();

    it('render', () => {
        expect(wrapper.find('.Pagination').exists()).toBe(true);
    });

    it('activePage', () => {
        expect(wrapper.find('.page-item.active').text()).toBe('1');
    });

    it('ellipsis', () => {
        expect(wrapper.find('.fa.fa-ellipsis-h').exists()).toBe(true);
    });

    // it('totalPage', () => {
    //     wrapper.find('.page-item').at(3).simulate('click');
    //     expect(wrapper.find('.page-item.active').text()).toBe('1');
    //     expect(props.onSelect).toHaveBeenCalled();
    //     // expect(wrapper.find('.page-item').at(wrapper.find('.page-item').length - 3).text()).toBe('10');
    // });

    it('ui', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
