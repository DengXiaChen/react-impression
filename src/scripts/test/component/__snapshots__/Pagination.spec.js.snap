// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination ui 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <ul
    className="Pagination"
    onSelect={[Function]}
    scope={3}
>
    <li
        className="page-item disabled"
    >
        <a
            className="page-link"
            href="javascript:void(0);"
            onClick={[Function]}
        >
            <i
                className="fa fa-angle-left"
            />
        </a>
    </li>
    <li
        className="page-item active"
    >
        <a
            className="page-link"
            href="javascript:void(0);"
            onClick={[Function]}
        >
            1
        </a>
    </li>
    <li
        className="page-item"
    >
        <a
            className="page-link"
            href="javascript:void(0);"
            onClick={[Function]}
        >
            2
        </a>
    </li>
    <li
        className="page-item"
    >
        <a
            className="page-link"
            href="javascript:void(0);"
            onClick={[Function]}
        >
            3
        </a>
    </li>
    <li
        className="page-item"
    >
        <a
            className="page-link"
            href="javascript:void(0);"
            onClick={[Function]}
        >
            4
        </a>
    </li>
    <li
        className="page-item disabled"
    >
        <i
            className="fa fa-ellipsis-h"
        />
    </li>
    <li
        className="page-item"
    >
        <a
            className="page-link"
            href="javascript:void(0);"
            onClick={[Function]}
        >
            <i
                className="fa fa-angle-right"
            />
        </a>
    </li>
</ul>,
  "nodes": Array [
    <ul
      className="Pagination"
      onSelect={[Function]}
      scope={3}
>
      <li
            className="page-item disabled"
      >
            <a
                  className="page-link"
                  href="javascript:void(0);"
                  onClick={[Function]}
            >
                  <i
                        className="fa fa-angle-left"
                  />
            </a>
      </li>
      <li
            className="page-item active"
      >
            <a
                  className="page-link"
                  href="javascript:void(0);"
                  onClick={[Function]}
            >
                  1
            </a>
      </li>
      <li
            className="page-item"
      >
            <a
                  className="page-link"
                  href="javascript:void(0);"
                  onClick={[Function]}
            >
                  2
            </a>
      </li>
      <li
            className="page-item"
      >
            <a
                  className="page-link"
                  href="javascript:void(0);"
                  onClick={[Function]}
            >
                  3
            </a>
      </li>
      <li
            className="page-item"
      >
            <a
                  className="page-link"
                  href="javascript:void(0);"
                  onClick={[Function]}
            >
                  4
            </a>
      </li>
      <li
            className="page-item disabled"
      >
            <i
                  className="fa fa-ellipsis-h"
            />
      </li>
      <li
            className="page-item"
      >
            <a
                  className="page-link"
                  href="javascript:void(0);"
                  onClick={[Function]}
            >
                  <i
                        className="fa fa-angle-right"
                  />
            </a>
      </li>
</ul>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <Pagination
        activePage={1}
        ellipsis={true}
        onSelect={[Function]}
        scope={3}
        totalPage={10}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Pagination {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "goPageHandle": [Function],
        "nextPageHandle": [Function],
        "prevPageHandle": [Function],
        "props": Object {
          "activePage": 1,
          "ellipsis": true,
          "onSelect": [Function],
          "scope": 3,
          "totalPage": 10,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <ul
          className="Pagination"
          onSelect={[Function]}
          scope={3}
>
          <li
                    className="page-item disabled"
          >
                    <a
                              className="page-link"
                              href="javascript:void(0);"
                              onClick={[Function]}
                    >
                              <i
                                        className="fa fa-angle-left"
                              />
                    </a>
          </li>
          <li
                    className="page-item active"
          >
                    <a
                              className="page-link"
                              href="javascript:void(0);"
                              onClick={[Function]}
                    >
                              1
                    </a>
          </li>
          <li
                    className="page-item"
          >
                    <a
                              className="page-link"
                              href="javascript:void(0);"
                              onClick={[Function]}
                    >
                              2
                    </a>
          </li>
          <li
                    className="page-item"
          >
                    <a
                              className="page-link"
                              href="javascript:void(0);"
                              onClick={[Function]}
                    >
                              3
                    </a>
          </li>
          <li
                    className="page-item"
          >
                    <a
                              className="page-link"
                              href="javascript:void(0);"
                              onClick={[Function]}
                    >
                              4
                    </a>
          </li>
          <li
                    className="page-item disabled"
          >
                    <i
                              className="fa fa-ellipsis-h"
                    />
          </li>
          <li
                    className="page-item"
          >
                    <a
                              className="page-link"
                              href="javascript:void(0);"
                              onClick={[Function]}
                    >
                              <i
                                        className="fa fa-angle-right"
                              />
                    </a>
          </li>
</ul>,
        "_debugID": 2,
        "_renderedOutput": <ul
          className="Pagination"
          onSelect={[Function]}
          scope={3}
>
          <li
                    className="page-item disabled"
          >
                    <a
                              className="page-link"
                              href="javascript:void(0);"
                              onClick={[Function]}
                    >
                              <i
                                        className="fa fa-angle-left"
                              />
                    </a>
          </li>
          <li
                    className="page-item active"
          >
                    <a
                              className="page-link"
                              href="javascript:void(0);"
                              onClick={[Function]}
                    >
                              1
                    </a>
          </li>
          <li
                    className="page-item"
          >
                    <a
                              className="page-link"
                              href="javascript:void(0);"
                              onClick={[Function]}
                    >
                              2
                    </a>
          </li>
          <li
                    className="page-item"
          >
                    <a
                              className="page-link"
                              href="javascript:void(0);"
                              onClick={[Function]}
                    >
                              3
                    </a>
          </li>
          <li
                    className="page-item"
          >
                    <a
                              className="page-link"
                              href="javascript:void(0);"
                              onClick={[Function]}
                    >
                              4
                    </a>
          </li>
          <li
                    className="page-item disabled"
          >
                    <i
                              className="fa fa-ellipsis-h"
                    />
          </li>
          <li
                    className="page-item"
          >
                    <a
                              className="page-link"
                              href="javascript:void(0);"
                              onClick={[Function]}
                    >
                              <i
                                        className="fa fa-angle-right"
                              />
                    </a>
          </li>
</ul>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Pagination
    activePage={1}
    ellipsis={true}
    onSelect={[Function]}
    scope={3}
    totalPage={10}
/>,
}
`;
